elastic-buffer-sim/src/Bittide/Simulate.hs

```diff
@@ -90,12 +93,38 @@ type HasUnderflowed = Bool
 type HasOverflowed = Bool
 type DisableTilHalf = Bool
 
+elasticBufferXilinx ::
+  forall readDom writeDom.
```

Comment:
You're currently making a wrapper around the Xilinx FIFO to make it act like `elasticBuffer` (including half-way logic n stuff), but I think we should flip this upside-down. Make the model behave like a very simple FIFO (including starting a zero), and move all the half-way logic to a wrapper.